src/yoroi-wallets/cardano/transactionManager.ts

```diff
@@ -0,0 +1,69 @@
+import {Storage} from '../storage'
+import {BackendConfig, Transactions} from '../types'
+import {TransactionCache} from './shelley'
+
+export const makeTransactionManager = async (storage: Storage, backendConfig: BackendConfig) => {
+  const transactionCache = await TransactionCache.create(storage.join('txs/'))
+  const memosManager = await makeMemosManager(storage.join('memos/'))
+
+  return {
+    // transactionCache api
+    getTransactions() {
+      const memos = memosManager.getMemos()
+      return Object.entries(transactionCache.transactions).reduce(
+        (result, [txId, tx]) => ({
+          ...result,
+          [txId]: {
+            ...tx,
+            memo: memos[txId] ?? null,
+          },
+        }),
+        {} as Transactions,
+      )
+    },
+    getPerRewardAddressCertificates() {
```

Comment:
How to expose these values?

Reply:
?
```
perAddressTxs() {
  return  transactionCache.perAddressTxs
}
```

Reply:
i would like to test this before push. Maybe it will happen like it happened here https://github.com/Emurgo/yoroi-mobile/pull/2306#discussion_r1090826052

Reply:
some apis are down

Reply:
It needs to be a getter to be able to get the updated values. Tested locally using a similar approach as in this branch https://github.com/Emurgo/yoroi-mobile/commit/2ac2021cd0d07f8fe4ec1668dc77188fd82f3181